lumibot/backtesting/backtesting_broker.py

```diff
@@ -557,39 +550,39 @@ def process_pending_orders(self, strategy):
             else:
                 continue
 
-    def limit_order(self, limit_price, side, close, high, low):
+    def limit_order(self, limit_price, side, open_, high, low):
         """Limit order logic."""
-        if side == "buy":
-            if limit_price >= close:
-                return close
-            elif limit_price < close and limit_price >= low:
-                return limit_price
-            elif limit_price < low:
-                return None
-        elif side == "sell":
-            if limit_price <= close:
-                return close
-            elif limit_price > close and limit_price <= high:
-                return limit_price
-            elif limit_price > high:
-                return None
-
-    def stop_order(self, stop_price, side, close, high, low):
+        # Gap Up case: Limit wasn't triggered by previous candle but current candle opens higher, fill it now
+        if side == "sell" and limit_price <= open_:
+            return open_
+
+        # Gap Down case: Limit wasn't triggered by previous candle but current candle opens lower, fill it now
+        if side == "buy" and limit_price >= open_:
+            return open_
+
+        # Current candle triggered limit normally
+        if low <= limit_price <= high:
+            return limit_price
+
+        # Limit has not been met
+        return 0.0
```

Comment:
should return None, not 0. 0 can actually be a fill price is some extreme situations